preload/Makefile

```diff
@@ -42,31 +43,42 @@ $(D)/libsb2.$(SHLIBEXT): LIBS := -ldl $(LIBSB2_LIBS)
 
 targets := $(targets) $(D)/libsb2.$(SHLIBEXT)
 
-$(D)/libsb2.o $(D)/sb_l10n.o: preload/exported.h
-$(D)/exported.h $(D)/ldexportlist: preload/wrappers.c
-$(D)/wrappers.c: preload/interface.master preload/gen-interface.pl
+$(D)/libsb2.o $(D)/sb_l10n.o: $(OBJDIR)/preload/exported.h
+preloader_generated_public := \
+	$(OBJDIR)/$(D)/wrappers.c \
+	$(OBJDIR)/$(D)/exported.h \
+	$(OBJDIR)/$(D)/syscallgate.c \
+	$(OBJDIR)/$(D)/ldexportlist \
+	$(OBJDIR)/$(D)/export.map \
+	$(OBJDIR)/$(D)/libsb2_interface.7
+
+$(preloader_generated_public): preload/interface.master \
+			preload/gen-interface.pl
 	$(MKOUTPUTDIR)
 	$(P)PERL
 	$(Q)$(SRCDIR)/preload/gen-interface.pl \
 		-n public \
-		-W preload/wrappers.c \
-		-E preload/exported.h \
-		-M preload/export.map \
+		-W $(OBJDIR)/preload/wrappers.c \
+		-E $(OBJDIR)/preload/exported.h \
+		-M $(OBJDIR)/preload/export.map \
 		-V $(PACKAGE_VERSION) \
 		-m preload/libsb2_interface.7 \
 		<$(SRCDIR)/preload/interface.master
 
-$(D)/privatewrappers.h: preload/privatewrappers.c
-$(D)/privatewrappers.c: preload/interface.private preload/gen-interface.pl
+preloader_generated_private := \
+	$(OBJDIR)/$(D)/privatewrappers.h \
+	$(OBJDIR)/$(D)/privatewrappers.c
+
+$(preloader_generated_private): preload/interface.private \
+			preload/gen-interface.pl
 	$(MKOUTPUTDIR)
 	$(P)PERL
 	$(Q)$(SRCDIR)/preload/gen-interface.pl \
 		-n private \
-		-W preload/privatewrappers.c \
-		-E preload/privatewrappers.h \
+		-W $(OBJDIR)/preload/privatewrappers.c \
+		-E $(OBJDIR)/preload/privatewrappers.h \
 		<$(SRCDIR)/preload/interface.private
 
-generated := preload/wrappers.c preload/exported.h preload/ldexportlist
-.PRECIOUS: $(generated)
+.PRECIOUS: $(prelader_generated_public) $(prelader_generated_private)
 
-CLEAN_FILES += $(generated)
+CLEAN_FILES += $(prelader_generated_public) $(prelader_generated_private)
```

Comment:
Typos in this and above, prelader_generated_public should be preloader_generated_public

Reply:
oh fu thanks